tests/unit/test_term_set.py

```diff
@@ -101,3 +103,86 @@ def test_folder_output(self):
+        self.wrapped_list = TermSetWrapper(value=['Homo sapiens'], termset=self.termset)
+
+        self.np_data = VectorData(
+            name='Species_1',
+            description='...',
+            data=self.wrapped_array
+        )
+        self.list_data = VectorData(
+            name='Species_1',
+            description='...',
+            data=self.wrapped_list
+        )
+
+    def test_properties(self):
+        self.assertEqual(self.wrapped_array.value, ['Homo sapiens'])
+        self.assertEqual(self.wrapped_array.termset.view_set, self.termset.view_set)
+        self.assertEqual(self.wrapped_array.dtype, 'U12') # this covers __getattr__
+
+    def test_get_item(self):
+        self.assertEqual(self.np_data.data[0], 'Homo sapiens')
+
+    def test_validate_error(self):
+        with self.assertRaises(ValueError):
+            VectorData(name='Species_1',
+                       description='...',
+                       data=TermSetWrapper(value=['Missing Term'],
+                       termset=self.termset))
+
+    def test_wrapper_validate_attribute(self):
+        col1 = VectorData(
+            name='Species_1',
+            description=TermSetWrapper(value='Homo sapiens',
+                                       termset=self.termset),
+            data=['Human']
+        )
+        self.assertTrue(isinstance(col1.description, TermSetWrapper))
+
+    def test_wrapper_validate_dataset(self):
+        col1 = VectorData(
+            name='Species_1',
+            description='...',
+            data=TermSetWrapper(value=['Homo sapiens'],
+                                termset=self.termset)
+        )
+        self.assertTrue(isinstance(col1.data, TermSetWrapper))
+
+    def test_wrapper_append(self):
+        data_obj = VectorData(name='species', description='...', data=self.wrapped_list)
+        data_obj.append('Mus musculus')
+        self.assertEqual(data_obj.data.value, ['Homo sapiens', 'Mus musculus'])
+
+    def test_wrapper_append_error(self):
+        data_obj = VectorData(name='species', description='...', data=self.wrapped_list)
+        with self.assertRaises(ValueError):
+            data_obj.append('bad_data')
+
+    def test_wrapper_extend(self):
+        data_obj = VectorData(name='species', description='...', data=self.wrapped_list)
+        data_obj.extend(['Mus musculus'])
+        self.assertEqual(data_obj.data.value, ['Homo sapiens', 'Mus musculus'])
+
+    def test_wrapper_extend_error(self):
+        data_obj = VectorData(name='species', description='...', data=self.wrapped_list)
+        with self.assertRaises(ValueError):
+            data_obj.extend(['bad_data'])
+
+    def test_wrapper_next(self):
+        data_obj = VectorData(name='species', description='...', data=self.wrapped_list)
+        data_obj.extend(['Mus musculus'])
+        it = iter(data_obj.data)
+        self.assertEqual(next(it), "Homo sapiens")
+        self.assertEqual(next(it), "Mus musculus")
```

Comment:
We can remove this too.
```suggestion
```

Reply:
This ended up not calling `__next__`. We would need to wrap an iterator which is currently not supported.